docs-src/concepts/what-is-key-management.md

```diff
@@ -0,0 +1,59 @@
+---
+id: what-is-key-management
+title: What is Key Management
+sidebar_label: Key Management
+description: Key Management is a necessary component of an encryption solution
+tags:
+  - keys
+  - encryption
+  - secrets
+---
+
+Key Management is a fundamental part of working with encryption keys. There are
+many computational and logistical aspects to generating and rotating keys, and
+this usually calls for a dedicated application in your stack. Here are some
+general recommendations for working with encryption keys in Temporal:
+
+- [Symmetric Encryption](https://en.wikipedia.org/wiki/Symmetric-key_algorithm)
+  is generally faster and will produce smaller payloads than asymmetric.
+  Normally, an advantage of _asymmetric_ encryption is that it allows you to
+  distribute your encryption and decryption keys separately, but depending on
+  your infrastructure, this may not offer any security benefits with Temporal.
+
+- AES-based algorithms are [hardware accelerated in Go](https://pkg.go.dev/crypto/aes) and other languages. AES algorithms are
+  widely vetted and trusted, and there are many different variants that may suit
+  your requirements. Load tests using `ALG_AES_256_GCM_HKDF_SHA512_COMMIT_KEY`
+  have performed well.
+
+- Store your encryption keys in the same manner as you store passwords, config
+  details, and other sensitive data. When possible, load the key into your
+  application, so you don't need to make a network call to retrieve it.
+  separate keys for each environment or namespace as much as possible.
+
+- **Make sure** you have a key rotation strategy in place in the event that your
+  keys are compromised or need to be replaced for another reason. Consider using
+  a dedicated secrets engine or a key management system (KMS).
+
+### Key Rotation
```

Comment:
We need to call out that for Temporal workflows in flight, old and new keys needs to be both accessible so that the existing workflows which were started with the old key can continue to make progress until completion.
This requires some planning and design especially if there are very long running workflows.